### PR TITLE
What is Wechaty sub-section of the  Introduction section

### PR DESCRIPTION
**Pull request by nibble0101**
# Improve the Introduction and Explanation sections project

| Section | Sub-section | Technical Writer |
| --- | --- | --- |
| Introduction | What is Wechaty | Joseph Mawa |

This  is **What is Wechaty** sub-section of the **Introduction** section

image
image
image
image
image
image

### CODE REVIEW

File: docusaurus/docs/what-is-wechaty.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+---
+
+import ProviderBadges   from './puppet-providers/transclusions/badges.mdx'
+import PolyglotBadges   from './polyglot/transclusions/badges.mdx'
+
+![Wechaty: Conversational RPA SDK for Chatbot Makers](/img/wechaty-logo.svg)
+
+## Introduction to Wechaty
+
+A chatbot is a software application used to conduct an online chat conversation instead of direct contact with a human being. Wechaty is an [Open Source](https://opensource.com/resources/what-open-source) software application for building chatbots. It is a modern Conversational [RPA](#placeholder) [SDK](#placeholder) which Chatbot makers can use to create a bot in a few lines of code. You can use Wechaty to build a chatbot which automates conversations and interact with people through instant messaging platforms such as [Whatsapp](https://www.whatsapp.com/?lang=en), [Wechat](https://www.wechat.com/), [Gitter](https://gitter.im/) and [Lark](https://www.larksuite.com/) among others.
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: I love how you covered a lot of information in the introduction. But maybe next time, we could try to eliminate the use of subordinate clauses :) See [Reduce subordinate clauses](https://developers.google.com/tech-writing/one/short-sentences#reduce_subordinate_clauses_optional)

File: docusaurus/docs/what-is-wechaty.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+- **Message processing**: You can use Wechaty to receive and send messages. It supports messages in the form of text, image, audio, video, and attachements.
+- **Room management**: You can use it to create rooms, add and remove room members, manage topics, etc
+- **Contact management**: Search by name, alias, tags, get profile data and avatar
+- **Friendship management**: Search and add new friends, accept friendship requests, etc
+- **Intelligent dialogue Management**: Just several configuration can get a task-oriented bot.
+- **Multi-platform support**: With Wechaty, you write code which runs on all IM platforms!
+
+## Supported Instant Messaging Services
+
+Wechaty supports a number of Instant Messaging platforms. You can build a chatbot using Wechaty for any of the following Instant Messaging platforms.
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: Please change the period at the end of the sentence to a colon. Thx!
(FYI [Lists](https://developers.google.com/style/lists))

File: docusaurus/docs/what-is-wechaty.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- [Wechat4U](#placeholder)
+- [Service](#placeholder)
+- [Mock](#placeholder)
+- [DIY](#placeholder)
+
+<ProviderBadges />
+
+## Supported Programming Languages
+
+The Wechaty community have developed [Software Development Kits](#placeholder) for most of the popular programming languages. You can build a Wechaty Chatbot in any of the programming languages listed below.
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: Please change the period at the end of the sentence to a colon. Thx!
(FYI [Lists](https://developers.google.com/style/lists))

File: docusaurus/docs/what-is-wechaty.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## Features of Wechaty
+
+Wechaty offers out-of-the-box support for the common features you would need for turning your [Instant Messaging](#placeholder) account into a chatbot. You can easily customize and extend Wechaty to create a chatbot that is tailored to your needs. Some common features of Wechaty that you can take advantage of include:
+
+<!-- Rethink these features -->
+
+- **Message processing**: You can use Wechaty to receive and send messages. It supports messages in the form of text, image, audio, video, and attachements.
+- **Room management**: You can use it to create rooms, add and remove room members, manage topics, etc
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: Please see the use of *etc*. [Word list](https://developers.google.com/style/word-list#letter-e)
image

File: docusaurus/docs/what-is-wechaty.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+- [Lark](https://www.larksuite.com/)
+- [Padlocal](#placeholder)
+- [Wechat4U](#placeholder)
+- [Service](#placeholder)
+- [Mock](#placeholder)
+- [DIY](#placeholder)
+
+<ProviderBadges />
+
+## Supported Programming Languages
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: Please capitalize only the first word in the title. See [Capitalization in titles and headings](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings)

File: docusaurus/docs/what-is-wechaty.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+<!-- Add Wechaty community as separate subsection under Introduction -->
+
+## Getting help
+
+Wechaty has a community of very helpful contributors on different platforms you can join to get help from. Before joining any of the communities, we recommend you read our [Code of conduct](#placeholder) so that you adhere to our community guidelines. A full list of the different Wechaty communities can be accessed from the [Wechaty community](#placeholder) section of this documentation.
+
+## Mission
+
+Our mission is to Make it easy for developers to build Chatbots using WeChaty
```
Comment (proudofsimin) [inline docusaurus/docs/what-is-wechaty.md]: It seems that you accidentally capitalized *M* in *make it easy* and forgot the period at the end.

**Review by proudofsimin — CHANGES_REQUESTED** [review]
Overall your documentation is really great! I point out some minor corrections and hope we could make it better together!

**Comment by huan** [comment]
Thanks to @proudofsimin for your great review & suggestions of how to follow the Google Developer Documentation Style Guide!

It's very important for our docs to follow those guides because we must be consistent across all docs, especially we have many writers work together.

I'm thinking about whether we can find a linter (linting tool) for enforce those documentation styles, would love to add it to our CI system if there's any.

**Comment by iamrajiv** [comment]
Hi @huan, I don't think there is a linter that follows the Google developer documentation style guide, but for programming languages, there are separate linters, but I don't think we need that.

So we have to make our linter according to the Google developer documentation style guide, but that would be very hard and can be considered a long process. So as of now, I would suggest making a separate markdown file that lists all the style guide to follow for Wechaty.

In addition to it, I want to add some miscellaneous points which can make the Wechaty workflow better:

- Adding linter to removed unnecessary indentations across all the docs file. There are so many files that have indentations like this (highlighted)

Screenshot 2021-05-23 at 8 05 56 PM

- Adding [Prettier](https://prettier.io) support for markdown files. Prettier is an opinionated code formatter. It removes all original styling and ensures that all outputted code conforms to a consistent style. (See this [blog post](https://archive.jlongster.com/A-Prettier-Formatter)).
- Adding spell check GitHub action.
- Adding GitHub action for translating Non-English issues content to English. For example, there are Non-English issues like [#161](https://github.com/wechaty/wechaty.js.org/issues/161). This action will convert that to English.

**Comment by proudofsimin** [comment]
> Thanks to @proudofsimin for your great review & suggestions of how to follow the Google Developer Documentation Style Guide!
> 
> It's very important for our docs to follow those guides because we must be consistent across all docs, especially we have many writers work together.
> 
> I'm thinking about whether we can find a linter (linting tool) for enforce those documentation styles, would love to add it to our CI system if there's any.

Hi @huan, 

Very great suggestions! but unfortunately I didn't find it. 
Like Rajiv suggested, if we could import grammar or spelling check, that would be awesome.

**Review by proudofsimin — APPROVED** [review]
fantastic!

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
Looks good to me, great work

**Comment by Rohitesh-Kumar-Jain** [comment]
ping @huan

**Comment by lijiarui** [comment]
hi @Rohitesh-Kumar-Jain As GSoD Project PR Review Rules in https://github.com/wechaty/wechaty.js.org/issues/1045

This PR should follow our rules and get enough approval, then I will review this PR. So I remove the `ready to merge` tag and feel free to add this tag when it gets enough approval.

**Comment by proudofsimin** [comment]
> hi @Rohitesh-Kumar-Jain As GSoD Project PR Review Rules in #1045
> 
> This PR should follow our rules and get enough approval, then I will review this PR. So I remove the `ready to merge` tag and feel free to add this tag when it gets enough approval.

Hi @chris-4444 Wechaty committee requested review from teammates. Could you review/ sign off it and add a tag `ready to merge` after you finish? Thank you.

**Review by chris-4444 — CHANGES_REQUESTED** [review]
Updated @nibble0101 @proudofsimin @Rohitesh-Kumar-Jain.

**Comment by Rohitesh-Kumar-Jain** [comment]
@chris-4444 I would like you to once go through the PR, and approve when you think it's ready to get merged.

**Review by chris-4444 — APPROVED** [review]
LGTM.

File: docusaurus/docs/what-is-wechaty.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+Since its creation in 2016, a number of Wechaty versions have been released. For more information about the release history and the current stable version, you can read the [Wechaty release notes](https://github.com/Wechaty/wechaty/releases) on Github.
+
+## License
+
+Wechaty is an [Open Source Project](https://opensource.com/resources/what-open-source). It is released under [Apache-2.0 license](https://github.com/wechaty/wechaty/blob/master/LICENSE) and the corresponding documentation is released under the [Creative Commons license](https://creativecommons.org/licenses/).
+
+## Maintainers
+
+Wechaty is maintained by [Huan](https://github.com/huan), [Rui](#placeholder) and a community of [Open Source Contributors](#placeholder). We are always looking for people to join the Wechaty community to maintain the Wechaty codebase and documentation. You necessarily don't have to be a programmer to contribute to Wechaty. To get started contributing, you can read the [Contributing section](#placeholder) of this documentation.
```
Comment (lijiarui) [inline docusaurus/docs/what-is-wechaty.md]: Here is the link:
```
[Rui](https://github.com/lijiarui)
[Open Source Commiters](https://github.com/wechaty/PMC#wechaty-committers)
[Open Source Contributors](https://wechaty.js.org/contributors/)
```

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: docusaurus/docs/what-is-wechaty.md (hunk 7 above)
Comment (nibble0101) [reply docusaurus/docs/what-is-wechaty.md]: Added the links

**Review by lijiarui — CHANGES_REQUESTED** [review]
I preview this doc but I found there are two `What is Wechaty here`
image

### 1. What is wechaty
image


### 2. What is wechaty
image

**Comment by nibble0101** [comment]
> I preview this doc but I found there are two `What is Wechaty here`
> image
> 
> ### 1. What is wechaty
> image
> 
> ### 2. What is wechaty
> image

The first one is what we wrote and the second is what was originally in the docs. I think we can remove the second one.

**Comment by lijiarui** [comment]
Hi @nibble0101 
> The first one is what we wrote and the second is what was originally in the docs. I think we can remove the second one.

I thinks instead of remove the origin part, you should merge them together, also remember to use mdx file here.

**Comment by nibble0101** [comment]
> Hi @nibble0101
> 
> > The first one is what we wrote and the second is what was originally in the docs. I think we can remove the second one.
> 
> I thinks instead of remove the origin part, you should merge them together, also remember to use mdx file here.

Well. I have merged the two files and changed it to `.mdx`

File: docusaurus/docs/wechaty.mdx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+title: What is Wechaty
+---
+
+import ShortestChatbots from './polyglot/transclusions/shortest-chatbots.mdx'
+import ProviderBadges from './puppet-providers/transclusions/badges.mdx'
+import PolyglotBadges from './polyglot/transclusions/badges.mdx'
+
+> A chatbot is a software application used to conduct an online chat conversation instead of direct contact with a live human agent. It is a type of software that can automate conversations and interact with people through messaging platforms. &mdash; [Wikipedia](https://en.wikipedia.org/wiki/Chatbot)
+
+Wechaty is an [Open Source](https://opensource.com/resources/what-open-source) software application for building chatbots. It is a modern Conversational RPA SDK which Chatbot makers can use to create a bot in a few lines of code.
```
Comment (lijiarui) [inline docusaurus/docs/wechaty.mdx]: Please add link (https://wechaty.js.org/docs/explainations/conversational) for Conversation

Please add link (http://wechaty.js.org/docs/explainations/rpa) for RPA

File: docusaurus/docs/wechaty.mdx [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+You can use Wechaty to build a chatbot which automates conversations and interact with people through instant messaging platforms such as [Whatsapp](https://www.whatsapp.com/?lang=en), [Wechat](https://www.wechat.com/), [Gitter](https://gitter.im/) and [Lark](https://www.larksuite.com/) among others.
```
Comment (lijiarui) [inline docusaurus/docs/wechaty.mdx]: Please add [Wecom](https://work.weixin.qq.com/) as well

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: docusaurus/docs/wechaty.mdx (hunk 8 above)
Comment (nibble0101) [reply docusaurus/docs/wechaty.mdx]: Changed the text to hyperlinks.

File: docusaurus/docs/wechaty.mdx (hunk 9 above)
Comment (nibble0101) [reply docusaurus/docs/wechaty.mdx]: Added Wecom to the list

**Review by huan — APPROVED** [review]
LGTM

@nibble0101 Thank you very much for this great PR!

@proudofsimin @Rohitesh-Kumar-Jain @chris-4444  Thank you very much for the review comments and the approvals.

This PR has been in the queue for a long time, and it has already fulfilled the requirements to be merged with three approvements:

1. Approved by the teammate (+1)
2. Approved by the volunteer team (+2)

Could you please help yourself to merge this PR for the community?

And I'd like to encourage all of our technical writers to merge a PR for the community, whenever the PR has fulfilled the requirements, and you believe it's good to be merged.

Thank you very much!

**Comment by nibble0101** [comment]
@Rohitesh-Kumar-Jain  Can you merge this? I have failed to merge because either the base branch is protected or because I am the one who opened the PR. I am not exactly sure why.

**Comment by huan** [comment]
I noticed that @lijiarui has required changes and it might be the blocker of merging this PR.

@nibble0101 Could you please resolve the change request from @lijiarui and ask her to approve this PR and try again to merge?

**Comment by nibble0101** [comment]
> I noticed that @lijiarui has required changes and it might be the blocker of merging this PR.
> 
> @nibble0101 Could you please resolve the change request from @lijiarui and ask her to approve this PR and try again to merge?

I have worked on the issues she raised and even requested another review from her.

**Comment by huan** [comment]
ping @lijiarui

File: docusaurus/docs/wechaty.mdx [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+import ShortestChatbots from './polyglot/transclusions/shortest-chatbots.mdx'
+import ProviderBadges from './puppet-providers/transclusions/badges.mdx'
+import PolyglotBadges from './polyglot/transclusions/badges.mdx'
+
+> A chatbot is a software application used to conduct an online chat conversation instead of direct contact with a live human agent. It is a type of software that can automate conversations and interact with people through messaging platforms. &mdash; [Wikipedia](https://en.wikipedia.org/wiki/Chatbot)
+
+Wechaty is an [Open Source](https://opensource.com/resources/what-open-source) software application for building chatbots. It is a modern [Conversational](explainations/conversational.md) [RPA](explainations/rpa.md) SDK which Chatbot makers can use to create a bot in a few lines of code.
+
+You can use Wechaty to build a chatbot which automates conversations and interact with people through instant messaging platforms such as [Whatsapp](https://www.whatsapp.com/?lang=en), [Wechat](https://www.wechat.com/), [Wecom](https://work.weixin.qq.com/), [Gitter](https://gitter.im/) and [Lark](https://www.larksuite.com/) among others.
```
Comment (lijiarui) [inline docusaurus/docs/wechaty.mdx]: Whatsapp --> WhatsApp
Wechat --> WeChat
Wecom --> WeCom

**Review by lijiarui — CHANGES_REQUESTED** [review]
Please change the IM name, then I will approve this PR. Thanks!

File: docusaurus/docs/wechaty.mdx (hunk 10 above)
Comment (nibble0101) [reply docusaurus/docs/wechaty.mdx]: Changed

**Review by lijiarui — APPROVED** [review]

**Pull request merged**